src/TwigComponent/src/Command/TwigComponentDebugCommand.php

```diff
@@ -0,0 +1,335 @@
+    {
+        $io = new SymfonyStyle($input, $output);
+        $name = $input->getArgument('name');
+        $directory = $input->getOption('dir');
+
+        if (\is_string($name)) {
+            $component = $this->findComponentName($io, $name, $input->isInteractive());
+            if (null === $component) {
+                $io->error(sprintf('Unknown component "%s".', $name));
+
+                return Command::FAILURE;
+            }
+
+            $this->displayComponentDetails($io, $component);
+
+            return Command::SUCCESS;
+        }
+
+        $components = $this->findComponents($directory);
+        $this->displayComponentsTable($io, $components);
+
+        return Command::SUCCESS;
+    }
+
+    public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
+    {
+        if ($input->mustSuggestArgumentValuesFor('name')) {
+            $suggestions->suggestValues(array_keys($this->findComponents()));
+        }
+    }
+
+    private function findComponentName(SymfonyStyle $io, string $name, bool $interactive): ?string
+    {
+        $components = [];
+        foreach ($this->componentClassMap as $componentName) {
+            if ($name === $componentName) {
+                return $name;
+            }
+            if (str_contains($componentName, $name)) {
+                $components[$componentName] = $componentName;
+            }
+        }
+
+        foreach ($this->findAnonymousComponents() as $componentName) {
+            if (isset($components[$componentName])) {
+                continue;
+            }
+            if ($name === $componentName) {
+                return $name;
+            }
+            if (str_contains($componentName, $name)) {
+                $components[] = $componentName;
+            }
+        }
+
+        if ($interactive && \count($components)) {
+            return $io->choice('Select one of the following component to display its information', array_values($components), 0);
+        }
+
+        return null;
+    }
+
+    /**
+     * @return array<string, ComponentMetadata>
+     */
+    private function findComponents(string $directory = null): array
+    {
+        $components = [];
+        foreach ($this->componentClassMap as $name => $class) {
+            $metadata = $this->componentFactory->metadataFor($name);
+            if (\is_string($directory) && !str_contains($metadata->getTemplate(), $directory)) {
+                continue;
+            }
+            $components[$name] ??= $metadata;
+        }
+        foreach ($this->findAnonymousComponents($directory) as $name => $template) {
+            $components[$name] ??= $this->componentFactory->metadataFor($name);
+        }
+
+        return $components;
+    }
+
+    /**
+     * Return a map of component name => template.
+     *
+     * @return array<string, string>
+     */
+    private function findAnonymousComponents(string $directory = null): array
```

Comment:
This argument is unused. And I think that makes sense (and I don't think the `dir` option makes sense anymore) now that we have a specific anonymous templates directory.

Reply:
I was not sure of the use cases, but i'm inclined to remove it too, and use the "name autocomplete/search" instead.